devnet-sdk/testing/systest/multi_client.go

```diff
@@ -283,15 +317,19 @@ func (mc *MultiClient) verifyFollowersWithRetry(
 				delete(pendingClients, clientIdx)
 			} else {
 				// Detected chain split - return error immediately without further retries
-				result.clientIndices = append(result.clientIndices, clientIdx)
-				result.hashes = append(result.hashes, hash)
+				hashMismatches.clientIndices = append(hashMismatches.clientIndices, clientIdx)
+				hashMismatches.hashes = append(hashMismatches.hashes, hash)
 				// Format and return chain split error
-				return result, formatChainSplitError(blockNum, primaryHash, clientIdx, hash)
+				return hashMismatches, formatChainSplitError(blockNum, primaryHash, clientIdx, hash)
 			}
 		}
 	}
 
-	return result, nil
+	// If we get here, we've exceeded retries
+	if len(pendingClients) == 0 {
+		return hashMismatches, nil
+	}
+	return hashMismatches, fmt.Errorf("exceeded max retries, %d clients still pending verification", len(pendingClients))
```

Comment:
```suggestion
	if len(pendingClients) == 0 {
		return hashMismatches, nil
	}
	// If we get here, we've exceeded retries
	return hashMismatches, fmt.Errorf("exceeded max retries, %d clients still pending verification", len(pendingClients))
```

Are you sure that this code is reachable? It looks to me like the loop above is designed to return early when we reach the max retries? It might be cleaner to panic if we get here and there are still pending clients, if we think it should be impossible. 

Perhaps we should add a unit test for this to exercise the different code paths.

Reply:
Added tests. Couldn't reach this part of the code. Yet when I read it, it seems possible. 🙃 
I'd love to add even more tests and simplify this code snippet, but I'm not familiar enough with the domain to feel confident making such a change.